spec/models/component_spec.rb

```diff
@@ -181,4 +181,185 @@ def empty_config(event_id)
+      expect(msa_component.sorted_certificates)
+        .to eql [msa_encryption, msa_secondary, msa_primary]
+    end
+
+    it 'orders components according lowest certificate present from (lowest to highest) days_left' do
+      zeus_msa = create(:msa_component, name: 'MSA Zeus')
+      zeus_sp = create(:sp_component, name: 'SP Zeus')
+      aphrodite_msa = create(:msa_component, name: 'MSA Aphrodite')
+      aphrodite_sp = create(:sp_component, name: 'SP Aphrodite')
+
+      zeus_sp_primary = create(:upload_certificate_event,
+        usage: CERTIFICATE_USAGE::SIGNING,
+        value: root.generate_encoded_cert(expires_in: 116.days),
+        component: zeus_sp).certificate
+      zeus_sp_secondary = create(:upload_certificate_event,
+        usage: CERTIFICATE_USAGE::SIGNING,
+        value: root.generate_encoded_cert(expires_in: 102.days),
+        component: zeus_sp).certificate
+      zeus_sp_encryption = encryption_certificate(component: zeus_sp, expires_in: 108.days)
+
+      aphrodite_sp_primary = create(:upload_certificate_event,
+        usage: CERTIFICATE_USAGE::SIGNING,
+        value: root.generate_encoded_cert(expires_in: 364.days),
+        component: aphrodite_sp).certificate
+      aphrodite_sp_secondary = create(:upload_certificate_event,
+        usage: CERTIFICATE_USAGE::SIGNING,
+        value: root.generate_encoded_cert(expires_in: 125.days),
+        component: aphrodite_sp).certificate
+      aphrodite_sp_encryption = encryption_certificate(expires_in: 130.days, component: aphrodite_sp)
+
+      aphrodite_msa_primary = create(:upload_certificate_event,
+        usage: CERTIFICATE_USAGE::SIGNING,
+        value: root.generate_encoded_cert(expires_in: 364.days),
+        component: aphrodite_msa).certificate
+      aphrodite_msa_secondary = create(:upload_certificate_event,
+        usage: CERTIFICATE_USAGE::SIGNING,
+        value: root.generate_encoded_cert(expires_in: 122.days),
+        component: aphrodite_msa).certificate
+      aphrodite_sp_encryption = encryption_certificate(expires_in: 140.days, component: aphrodite_msa)
+
+      zeus_msa_primary = create(:upload_certificate_event,
+        usage: CERTIFICATE_USAGE::SIGNING,
+        value: root.generate_encoded_cert(expires_in: 128.days),
+        component: zeus_msa).certificate
+      zeus_msa_secondary = create(:upload_certificate_event,
+        usage: CERTIFICATE_USAGE::SIGNING,
+        value: root.generate_encoded_cert(expires_in: 105.days),
+        component: zeus_msa).certificate
+      zeus_msa_encryption = encryption_certificate(component: zeus_msa, expires_in: 138.days)
+
+      travel_to Time.now + 100.days
+      components = MsaComponent.where(name: ['MSA Zeus','MSA Aphrodite']) + SpComponent.where(name: ['SP Zeus','SP Aphrodite'])
+
+      expect(components.sort_by(&:days_left)).to eq [zeus_sp, zeus_msa, aphrodite_msa, aphrodite_sp]
+      expect(components.sort_by(&:days_left).map(&:days_left)).to eq [2, 5, 22, 25]
+    end
+  end
```

Comment:
good test coverage... can we have one more tests exercising the `NON_SORTING_SEED` logic?

Reply:
Ok